app/src/main/java/nl/mpcjanssen/simpletask/util/Config.kt

```diff
@@ -257,63 +213,45 @@ object Config : SharedPreferences.OnSharedPreferenceChangeListener {
     val todoFile: File
         get() = File(todoFileName)
 
-    @SuppressLint("CommitPrefEdits")
     fun setTodoFile(todo: String) {
-        with(prefs.edit()) {
-            putString(getString(R.string.todo_file_key), todo)
-            remove(getString(R.string.file_current_version_id))
-            commit()
-        }
+        _todoFileName = todo
+        prefs.edit().remove(getString(R.string.file_current_version_id)).apply()
     }
 
     fun clearCache() {
         todoList = null
         currentVersionId = null
     }
 
-    val isAutoArchive: Boolean
-        get() = prefs.getBoolean(getString(R.string.auto_archive_pref_key), false)
+    val isAutoArchive by BooleanPreference(R.string.auto_archive_pref_key, false)
 
-    val hasPrependDate: Boolean
-        get() = prefs.getBoolean(getString(R.string.prepend_date_pref_key), true)
+    val hasPrependDate by BooleanPreference(R.string.prepend_date_pref_key, true)
 
-    val hasKeepSelection: Boolean
-        get() = prefs.getBoolean(getString(R.string.keep_selection), false)
+    val hasKeepSelection by BooleanPreference(R.string.keep_selection, false)
 
-    val hasKeepPrio: Boolean
-        get() = prefs.getBoolean(getString(R.string.keep_prio), true)
+    val hasKeepPrio by BooleanPreference(R.string.keep_prio, true)
 
-    val shareAppendText: String
-        get() = prefs.getString(getString(R.string.share_task_append_text), " +background")
+    val shareAppendText by StringPreference(R.string.share_task_append_text, " +background")
 
-    var latestChangelogShown: Int
-        get() = prefs.getInt(getString(R.string.latest_changelog_shown), 0)
-        set(versionCode) {
-            prefs.edit().putInt(getString(R.string.latest_changelog_shown), versionCode).commit()
-        }
+    var latestChangelogShown by IntPreference(R.string.latest_changelog_shown, 0)
 
-    val localFileRoot: String
-        get() = prefs.getString(getString(R.string.local_file_root), "/sdcard/")
+    val localFileRoot by StringPreference(R.string.local_file_root, "/sdcard/")
 
-    val hasCapitalizeTasks: Boolean
-        get() = isCapitalizeTasks
```

Comment:
Removed this because it was unused, and also just a proxy for `isCapitalizeTasks`, which is already public